fetch.js

```diff
@@ -217,6 +217,11 @@
 
   function Request(url, options) {
     options = options || {}
+
+    if (url && url.match(/^\w+?:\/\/\w+?:\w+?@/)) {
+      throw new TypeError('Credentials not allowed in request URL')
+    }
+
```

Comment:
It's a bummer that we need to pull in a dependency just to be able to raise exceptions in rare cases. I think that a regex such as this would be just fine:

```js
/^([\w-]+:)?\/\/[^\/]+@/
```